### PR TITLE
postgis: allow on Darwin

### LINKED ISSUE #65452: postgis: support macOS
**Describe the bug**
pkgs/servers/sql/postgresql/ext/postgis.nix has `platforms = platforms.linux`, preventing it from being used on other platforms, particularly Darwin in my case.

**To Reproduce**

1. Populate shell.nix:
```
{ pkgs ? import <nixpkgs> {} }:
pkgs.stdenv.mkDerivation {
  name="good";
  buildInputs = [ (pkgs.postgresql.withPackages (p: [ p.postgis ])) ];
}
```

2. Run nix-shell to get this error:

error: Package ‘postgis-2.5.2’ in /Users/lukeworth/.nix-defexpr/channels/nixpkgs/pkgs/servers/sql/postgresql/ext/postgis.nix:63 is not supported on ‘x86_64-darwin’, refusing to evaluate.

**Expected behavior**
Produce a postgresql-and-plugins package with postgis available.

**Additional context**
Simply allowing `platforms = platforms.all` doesn't work immediately because PostGIS' build system is quirky. There was some relevant-looking discussion on #38369.

**Metadata**
 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.7.0, macOS 10.14.6`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.1`
 - channels(lukeworth): `"nixpkgs-19.09pre185562.00ef72610c8"`
 - nixpkgs: `/Users/lukeworth/nixpkgs`

### PR DESCRIPTION
**Pull request by lrworth**
Resolves #65452.

To get PostGIS going on Darwin:
1. Add libiconv, as is often required.
2. Expand platforms to `platforms.all`.
3. Deal with PostGIS' quirky build system.

PostGIS' configure.ac has the following gem:

```
  AC_MSG_RESULT([------------------------------------------------------------------------])
  AC_MSG_RESULT([  WARNING: You have set the --prefix to '$prefix'. But we mostly    ])
  AC_MSG_RESULT([  ignore the --prefix. For your info, using the values determined from ])
  AC_MSG_RESULT([  $PG_CONFIG we will be installing:   ])
  AC_MSG_RESULT([    * postgis shared library in $PGSQL_LIBDIR ])
  AC_MSG_RESULT([    * postgis SQL files in $PGSQL_SHAREDIR/contrib/postgis-$POSTGIS_MAJOR_VERSION.$POSTGIS_MINOR_VERSION ])
  AC_MSG_RESULT([    * postgis executables in $PGSQL_BINDIR ])
  AC_MSG_RESULT([------------------------------------------------------------------------])
```

This is suggestive of some assumptions in the build system which are
revealed when building in Nix on Darwin: the build fails because the
postgres binary cannot be found in the install prefix specified for
postgis; viz.

```
  cc x -bundle_loader $POSTGIS_PREFIX/bin/postgres
```

The bundle_loader parameter is only available on Darwin `ld`, and this
problem doesn't appear to affect Linux systems.

The solution presented here is to symlink the postgres binary where
PostGIS expects it to be, and then remove it after the build completes.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by vcunat** [comment]
@GrahamcOfBorg build postgis

**Comment by lrworth** [comment]
`gdal` doesn't build with sandbox enabled. I don't know how to debug this problem right now.

**Comment by vcunat** [comment]
Borg's build did succeed on Darwin.  I actually don't remember what's status of the Darwin sandbox.  I assume further improvements can be left for later (in any case I'd leave some time for others to react).

File: pkgs/servers/sql/postgresql/ext/postgis.nix [hunk 1]
```diff
@@ -43,14 +45,13 @@ stdenv.mkDerivation rec {
...
-  '';
-
-  preInstall = ''
     mkdir -p $out/bin
+    ln -s ${postgresql}/bin/postgres $out/bin/postgres
   '';
 
   # create aliases for all commands adding version information
   postInstall = ''
+    rm $out/bin/postgres
```
Comment (kalbasit) [inline pkgs/servers/sql/postgresql/ext/postgis.nix]: What's the reason for linking Postgres in postConfigure only to remove it in postInstall?

File: pkgs/servers/sql/postgresql/ext/postgis.nix [hunk 2]
```diff
@@ -43,14 +45,13 @@ stdenv.mkDerivation rec {
...
     for prog in $out/bin/*; do # */
```
Comment (kalbasit) [inline pkgs/servers/sql/postgresql/ext/postgis.nix]: Although this is not a change in this PR, do you know why there's a `*/` comment here?

File: pkgs/servers/sql/postgresql/ext/postgis.nix (hunk 1 above)
Comment (kalbasit) [reply pkgs/servers/sql/postgresql/ext/postgis.nix]: Never mind, I just saw the reason in the PR description :man_facepalming:. Perhaps a comment in the code to highlight the issue?

File: pkgs/servers/sql/postgresql/ext/postgis.nix (hunk 2 above)
Comment (vcunat) [reply pkgs/servers/sql/postgresql/ext/postgis.nix]: That clearly seems to unbreak syntax highlighting in editors that get confused by `/*` on that same line.  One can see similar ones all over nixpkgs.

File: pkgs/servers/sql/postgresql/ext/postgis.nix [hunk 3]
```diff
@@ -64,6 +65,6 @@ stdenv.mkDerivation rec {
     homepage = https://postgis.net/;
     license = licenses.gpl2;
     maintainers = [ maintainers.marcweber ];
-    platforms = platforms.linux;
+    platforms = platforms.all;
```
Comment (marsam) [inline pkgs/servers/sql/postgresql/ext/postgis.nix]:
```suggestion
    platforms = postgresql.meta.platforms;
```

Comment (kalbasit) [reply pkgs/servers/sql/postgresql/ext/postgis.nix]:
```suggestion
    inherit (postgresql.meta) platforms;
```

also works.

**Review by marsam — APPROVED** [review]

**Review by kalbasit — APPROVED** [review]

**Comment by kalbasit** [comment]
@GrahamcOfBorg build postgis

**Pull request merged**